src/main/java/core/basesyntax/dao/ActivityDaoCsvFileImpl.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.Activity;
+import core.basesyntax.util.Util;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ActivityDaoCsvFileImpl implements ActivityDao {
+    private static final long TITLE_LINE = 1;
+    private final String inputFile;
+
+    public ActivityDaoCsvFileImpl(String inputFile) {
+        this.inputFile = inputFile;
+    }
+
+    @Override
+    public List<Activity> getAll() {
+        List<String> strings = Util.readFile(inputFile);
+        return strings.stream()
+                .skip(TITLE_LINE)
+                .map(this::parseActivity)
+                .collect(Collectors.toList());
```

Comment:
dao should not contain any business logic. Let only your repository return here. You can put this business logic in your parser.

Reply:
Dao doesn't know anything about service.